### PR TITLE
Weather App - Project 6

### PR DESCRIPTION
**Pull request by Andrea-Osmar**
Link to my deployed project: https://unicornweather.netlify.app/

My reflections on how this week's project turned out: it turned out alright.. I am very proud that i actually managed to get this project done. However I know that the sunrise & sunset time is swedish time nomather what city you choose. There is also something a little off with the weather container on my phone but not in the dev tools :/ will try to figure that out
I know that i have not commited very much but I had some issues with my internet and GIthub..

Things I'd like to have clarified or explained in more detail: help with sunrise & sunset time. I am still very unsure on how fetch and promises work..

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            objectDate[date].push(item);
+        } else {
+            objectDate[date] = [item];
+        }
+    });
+
+    // AT 12.00
+    const filteredForecastNoon = json.list.filter((item) =>
+        item.dt_txt.includes("12:00:00"));
+    filteredForecastNoon.forEach((forecast) => {
```
Comment (karolinalbinsson) [inline script.js:195]: I think you could work with your indentation in some of your functions, like this one. It is a bit of a struggle to read it.

File: script.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    let weatherContainerWidth = weatherContainer.clientWidth;
+
+    // POSITION VISIBLE & IN MIDDLE OF SCREEN
+    weatherContainer.style.left = `calc(50% - ${weatherContainerWidth / 2}px)`;
+    weatherContainer.style.top = `calc(50% - ${weatherContainerHeight / 1.5}px)`;
+    weatherContainer.style.visibility = "visible";
+};
+
+document.getElementById("searchBtn").addEventListener("click", () => {
+    let searchTerm = document.getElementById("searchInput").value;
```
Comment (karolinalbinsson) [inline script.js:228]: Consider changing this to a const since the value of searchTerm is never changed in your function

File: script.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    sunsetTimeElement.innerHTML = "Sunset: " + sunsetTime;
+    setPositionForWeatherInfo();
+};
+
+// FORECAST FOR 5 DAYS
+const readForecast = (json) => {
+    const forecastContainer = document.getElementById("forecast");
+    const forecastContainer2 = document.getElementById("forecast2");
+
+    // REMOVE FORECAST WHEN CHANGING CITY
```
Comment (karolinalbinsson) [inline script.js:159]: Good that you have put a comment here, so I understand what's going on :)

File: script.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+    const windSpeedElement = document.getElementById("windSpeed");
+
+    const humidityElement = document.getElementById("humidity");
+
+    const sunriseTimeElement = document.getElementById("sunrise");
+
+    const sunsetTimeElement = document.getElementById("sunset");
+
+    const CityHeader = document.getElementById("cityHeader");
```
Comment (karolinalbinsson) [inline script.js:107]: camelCase, I think you just forgot it here. :)

File: script.js [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        hour: "2-digit",
+        minute: "2-digit",
+    });
+
+    sunsetTimeElement.innerHTML = "Sunset: " + sunsetTime;
+    setPositionForWeatherInfo();
+};
+
+// FORECAST FOR 5 DAYS
+const readForecast = (json) => {
```
Comment (karolinalbinsson) [inline script.js:155]: This function is really big, could it be divided into multiple functions that would be called by this one?

File: script.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    weatherContainer.style.top = `calc(50% - ${weatherContainerHeight / 1.5}px)`;
+    weatherContainer.style.visibility = "visible";
+};
+
+document.getElementById("searchBtn").addEventListener("click", () => {
+    let searchTerm = document.getElementById("searchInput").value;
+    if (searchTerm) {
+        searchWeather(searchTerm);
+        searchForecast(searchTerm);
+    }
```
Comment (karolinalbinsson) [inline script.js:232]: I didn't receive any feedback from the app when I searched for a city that doesn't exist, maybe an else statement could fix that?

File: style.css [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+#documentIconImg {
+    -webkit-transform: scale(1);
+    transform: scale(1);
+    -webkit-transition: .3s ease-in-out;
+    transition: .3s ease-in-out;
+}
+
+#documentIconImg:hover {
```
Comment (karolinalbinsson) [inline style.css:78]: This animation is really cute! :+1:

File: script.js [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    // WEATHER DESCRIPTION FROM OPENWEATHER MAP
+    const resultDescription = resultFromServer.weather[0].description;
+    weatherDescriptionHeader.innerText =
+        resultDescription.charAt(0).toUpperCase() + resultDescription.slice(1);
+
+    temperatureElement.innerHTML =
+        Math.floor(resultFromServer.main.temp) + "&#176c";
+
+    windSpeedElement.innerHTML =
+        "Winds at " + Math.floor(resultFromServer.wind.speed) + " m/s";
```
Comment (karolinalbinsson) [inline script.js:126]: I suggest using string templating instead of + here:
`Winds at ${variable} m/s`

**Review by karolinalbinsson — CHANGES_REQUESTED** [review]
Hey Andrea!
Great job with the weather app! 

Some reflections:
I love your design and think the app is very pleasing to look at. The different images in the background depending of the weather gives your application a really nice touch. Also your choice of font and margins/padding makes it pleasant for the eye. 

When I read your own reflections of how you think the project turned out, I sense some disappointment in the text(?), but I really don't think you should be disappointed at all! You have done a great job with letting the user select a city, changing the appearance of the app depending of weather, and made it responsive as well. 

I struggled with getting the time in the correct format aswell, it has to do with the date objects and functions that are converting the time to whichever timezone is on the client system (I hope that whomever of the code coaches are doing this review can explain it a bit more.)

I have some remarks and I believe you need to make some small changes for fulfilling the general requirements of how to write our code.You can see my remarks in the comments on the files, but here are the ones that need action:

1. You do use a lot of styling by ID, and I think you should consider changing it to style by class name.
2. Also have a look at the section named Template literals here and use that approach: (https://www.notion.so/Guidelines-for-how-to-write-good-code-69ffb59eac4e4358a7c5752efb873c4e)
 
Overall - good job!! :+1: :1st_place_medal:

File: script.js [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    });
+
+    sunsetTimeElement.innerHTML = "Sunset: " + sunsetTime;
+    setPositionForWeatherInfo();
+};
+
+// FORECAST FOR 5 DAYS
+const readForecast = (json) => {
+    const forecastContainer = document.getElementById("forecast");
+    const forecastContainer2 = document.getElementById("forecast2");
```
Comment (HIPPIEKICK) [inline script.js:157]: What's the difference between these two? Could we create more descriptive variable names to make the difference clear?

File: style.css [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+.forecastDays {
+  padding-right: 30px;
+}
+
+/** MEDIA QUERIES**/
+
+@media (min-width: 668px) and (max-width: 1024px) {
+  #weatherContainer {
+    position: absolute;
```
Comment (HIPPIEKICK) [inline style.css:125]: Why `position: absolute;` here without positioning? It makes the container go to the far left side
Skärmavbild 2020-10-13 kl  11 42 57

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job Andrea, especially nice with the search function! Take a look at Karolin's comments and also the comment I made about `position: absolute;`.

Regarding the formatting of the dates, take a look at this [Slack-thread](https://technigo.slack.com/archives/C6QAN25EV/p1601018607030900) and this [Stack Overflow-question](https://stackoverflow.com/c/technigo/questions/1586).

File: style.css (hunk 10 above)
Comment (Andrea-Osmar) [reply style.css:125]: A question, on my computer the container never goes to the far left.
I positioned the element in my javascript, and I needed position absolute for that to work. 
Do I then also need to position the element in css?


  // POSITION VISIBLE & IN MIDDLE OF SCREEN
    weatherContainer.style.left = `calc(50% - ${weatherContainerWidth / 2}px)`;
    weatherContainer.style.top = `calc(50% - ${weatherContainerHeight / 1.5}px)`;
    weatherContainer.style.visibility = "visible";

Comment (HIPPIEKICK) [reply style.css:125]: That's really weird! I can't seem to recreate the issue now either. Sorry about that, approving!

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**